File: apps/api/app/schemas/account.py
from pydantic import BaseModel


class CreateAccountInput(BaseModel):
    name: str
    currency: str  # USD | PEN
    initialBalance: float
    color: str | None = "#3b82f6"


class UpdateAccountInput(BaseModel):
    name: str | None = None
    color: str | None = None
    initialBalance: float | None = None


class AccountResponse(BaseModel):
    id: str
    name: str
    currency: str
    initialBalance: float
    currentBalance: float
    color: str
    createdAt: str
    updatedAt: str

File: packages/types/src/account.ts
export type Currency = "USD" | "PEN";

export interface Account {
  id: string;
  name: string;
  currency: Currency;
  initialBalance: number;
  currentBalance: number;
  color: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateAccountInput {
  name: string;
  currency: Currency;
  initialBalance: number;
  color?: string;
}

export interface UpdateAccountInput {
  name?: string;
  color?: string;
  initialBalance?: number;
}
